src/dma.c

```diff
@@ -135,17 +183,17 @@ void dma_write_raw_async(const void * ram_address, unsigned long pi_address, uns
  * 
  * @note This function must be called with interrupts disabled.
  */
-static uint32_t __io_read32(unsigned long pi_address) {
+static uint32_t __io_read32(void* pi_pointer) {
```

Comment:
So this means whenever we access something from the CPU it should be a virtual address?

Reply:
Yes, CPU only access virtual addresses by definition. A virtual address is 3 bit segment + 29 bit address, and is mapped to a physical address either directly or indirectly via TLB.
See also: https://n64brew.dev/wiki/Memory_map